lib/sycamore/sycamore/query/planner.py

```diff
@@ -81,6 +80,104 @@ def process_json_plan(json_plan: str) -> LogicalPlan:
     return LogicalPlan.model_validate(parsed_plan)
 
 
+def postprocess_llm_helper(user_message: str, llm_client: LLM) -> str:
+    messages = [
+        {
+            "role": "system",
+            "content": """You are a helpful agent that assists in small transformations
+                    of input text as per the instructions. You should make minimal changes
+                    to the provided input and keep your response short""",
+        },
+        {"role": "user", "content": user_message},
+    ]
+
+    prompt_kwargs = {"messages": messages}
+    chat_completion = llm_client.generate(prompt_kwargs=prompt_kwargs, llm_kwargs={"temperature": 0, "seed": 42})
+    return chat_completion
+
+
+def postprocess_plan(plan: Any, llm_client: LLM) -> Any:
+    """Given the LogicalPlan query plan provided by the LLM, postprocess it using a set of rules
+    that modify the plan for optimizing or other purposes."""
+
+    # Rule 1: If the plan has a vector search in the beginning followed by a count or nothing or extract_entity,
+    # we replace that by removing the vector search and adding an LLM Filter before the count
+
+    if plan.nodes[0].node_type == "QueryVectorDatabase" and (
+        len(plan.nodes) == 1 or plan.nodes[1].node_type in ["Count", "LlmExtractEntity", "SummarizeData"]
+    ):
+        # If the first operator has an "opensearch_filter", we will convert it to a QueryDatabase with
+        #        that opensearch_filter as "query"
+        # If not, we do a QueryDatabase with "match_all" instead
+        op = plan.nodes[0]
+
+        modified_description = postprocess_llm_helper(
+            f"""
+                    The following is the description of a Python function. I am modifying the function code
+                    to remove any functionality that specifically has to do with "{op.query_phrase}", thereby 
+                    generalizing the description to be more flexible.
+                    Return only the modified description. Do not make assumptions
+                    about the intent of the question that are not explicitly specified.
+                    {op.description}
+                    """,
+            llm_client,
+        )
+
+        new_op = QueryDatabase.model_validate(
```

Comment:
curious why you're using model_validate and a dict as opposed to just create a regular object directly?

Reply:
Primarily to stay consistent with how they are being created in the other code right now and not duplicate functionality. Also, I didn't see an appropriate __init__ function, but perhaps it is being created implicitly? I can change it to direct object creation if that's the more appropriate way to do it.

Reply:
You should just be able to initialize them like so: https://github.com/aryn-ai/sycamore/blob/0e2a9f5e586474c15bbfd1f85cf8b3bc3a1e4543/lib/sycamore/sycamore/tests/unit/query/test_planner.py#L68

I had suggested since we're creating the object ourselves as opposed to trying to create one from another data source (like the llm giving us a json).

Don't block on it though, it's your call if trivial to do